### DISCUSSION TITLE
Dynamic IRequestHandler registration for all services

### DISCUSSION THREAD

**Discussion by stefan0s** [LuckyPennySoftware/MediatR#901] [Q&A]
Hello,

i want to integrate mediatr pipeline behaviours to apply caching and messaging functionality to my services. I am currently having my service class configured like this:
```
public class MyService : IMyService {
    private readonly IdbContext _db;
    public MyService(IdbContext db)
    {
        _db = db;
    }

    [Caching(60)]
    public async Task<string> Method1(int id)
    {
        return await _db.getData(id);
    }

    [Caching(20)]
    public async Task<string> Method2(int id, string name)
    {
        return await _db.getDataName(id, name);
    }
 }
```

I want to create an attribute to set that each method is cachable or not and set the cache timeout. Is there any way to dynamically register IRequestHandler for these methods?

**Comment by jbogard** [comment]
No. You might be able to use behaviors for this, however.

**Reply by stefan0s** [reply]
Hello jbogard,

How can I do that? Can you please share it?

**Reply by jbogard** [reply]
No it was just an idea, youll need to play around with it
